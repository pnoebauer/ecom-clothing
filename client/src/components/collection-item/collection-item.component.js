import React from 'react';

import { connect } from 'react-redux';

import {
	CollectionItemContainer,
	ImageContainer,
	AddButton,
	CollectionFooterContainer,
	NameContainer,
	PriceContainer
} from './collection-item.styles';

import { addItemToCart } from '../../redux/cart/cart.actions';

const CollectionItem = ({item, addItem}) => {
	const {name, price, imageUrl} = item;
	return (
		<CollectionItemContainer>
			<ImageContainer 
				className='image'
				style={{
						backgroundImage: `url(${imageUrl})`
					}}
			/>
			<CollectionFooterContainer>
				<NameContainer>{name}</NameContainer>
				<PriceContainer>{price}</PriceContainer>
			</CollectionFooterContainer>
			<AddButton inverted onClick={() => addItem(item)} >Add to Cart</AddButton>
		</CollectionItemContainer>
	);
};

const mapDispatchToProps = dispatch => {
	return {
		addItem: item => dispatch(addItemToCart(item))
	};
};

export default connect(null, mapDispatchToProps)(CollectionItem);