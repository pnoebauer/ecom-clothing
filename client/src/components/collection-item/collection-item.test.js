import React from 'react';
import {shallow} from 'enzyme';
import {CollectionItem} from './collection-item.component';
import {AddButton} from './collection-item.styles';

describe('CollectionItem component tests', () => {
	let wrapper;
	let mockAddItem;

	const mockItem = {
		imageUrl: 'www.testImage.com',
		price: 10,
		name: 'hats',
	};

	beforeEach(() => {
		mockAddItem = jest.fn();

		const mockProps = {
			item: mockItem,
			addItem: mockAddItem,
		};

		wrapper = shallow(<CollectionItem {...mockProps} />);
	});

	test('render CollectionItem component', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('call mockAddItem when the AddButton is clicked', () => {
		wrapper.find(AddButton).simulate('click');
		// expect(mockAddItem).toHaveBeenCalled();
		expect(mockAddItem).toHaveBeenCalledWith(mockItem);
	});
});
