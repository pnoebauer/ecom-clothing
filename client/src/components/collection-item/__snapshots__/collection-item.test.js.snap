// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component tests render CollectionItem component 1`] = `
<collection-itemstyles__CollectionItemContainer>
  <collection-itemstyles__ImageContainer
    className="image"
    style={
      Object {
        "backgroundImage": "url(www.testImage.com)",
      }
    }
  />
  <collection-itemstyles__CollectionFooterContainer>
    <collection-itemstyles__NameContainer>
      hats
    </collection-itemstyles__NameContainer>
    <collection-itemstyles__PriceContainer>
      10
    </collection-itemstyles__PriceContainer>
  </collection-itemstyles__CollectionFooterContainer>
  <collection-itemstyles__AddButton
    inverted={true}
    onClick={[Function]}
  >
    Add to Cart
  </collection-itemstyles__AddButton>
</collection-itemstyles__CollectionItemContainer>
`;
